File: flutter_myplugin/ios/Classes/FlutterMypluginPlugin.m
#import "FlutterMypluginPlugin.h"
#import "FlutterMyPluginEvent.h"

@implementation FlutterMypluginPlugin

FlutterMyPluginEvent *pluginEvent;

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_myplugin"
            binaryMessenger:[registrar messenger]];
  FlutterMypluginPlugin* instance = [[FlutterMypluginPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
    
    pluginEvent = [[FlutterMyPluginEvent alloc] init];
    pluginEvent.eventChannel = [FlutterEventChannel eventChannelWithName:@"flutter_plugin_event" binaryMessenger:[registrar messenger]];
    [pluginEvent.eventChannel setStreamHandler:pluginEvent];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"getNative" isEqualToString:call.method]) {
    result(@"Get iOS Success");
  } else if ([@"sayHello" isEqualToString:call.method]) {
      NSDictionary* argsMap = call.arguments;
      NSString* message = argsMap[@"message"];
//      NSLog(message);
    //result(message);//直接返回内容
      result(@{@"message":message,@"info":@"info_value"});
      
      FlutterEventSink eventSink = pluginEvent.eventSink;
      if(eventSink){
          eventSink(@{
              @"event": @"demoEvent",
              @"value": @"value is 10 ios"
          });
      }
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: flutter_myplugin/android/src/main/java/com/example/flutter_myplugin/FlutterMypluginPlugin.java
package com.example.flutter_myplugin;

import android.widget.Toast;

import androidx.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.EventChannel;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * FlutterMypluginPlugin
 */
public class FlutterMypluginPlugin implements FlutterPlugin, MethodCallHandler {
    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private MethodChannel channel;

    private EventChannel.EventSink eventSink = null;

    private EventChannel.StreamHandler streamHandler = new EventChannel.StreamHandler() {
        @Override
        public void onListen(Object arguments, EventChannel.EventSink sink) {
            eventSink = sink;
        }

        @Override
        public void onCancel(Object arguments) {
            eventSink = null;
        }
    };

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
        channel = new MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "flutter_myplugin");
        channel.setMethodCallHandler(this);

        EventChannel eventChannel = new EventChannel(flutterPluginBinding.getBinaryMessenger(),"flutter_plugin_event");
        eventChannel.setStreamHandler(streamHandler);
    }

    // This static function is optional and equivalent to onAttachedToEngine. It supports the old
    // pre-Flutter-1.12 Android projects. You are encouraged to continue supporting
    // plugin registration via this function while apps migrate to use the new Android APIs
    // post-flutter-1.12 via https://flutter.dev/go/android-project-migration.
    //
    // It is encouraged to share logic between onAttachedToEngine and registerWith to keep
    // them functionally equivalent. Only one of onAttachedToEngine or registerWith will be called
    // depending on the user's project. onAttachedToEngine or registerWith must both be defined
    // in the same class.
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_myplugin");
        channel.setMethodCallHandler(new FlutterMypluginPlugin());
    }

    @Override
    public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
        if (call.method.equals("getPlatformVersion")) {
            result.success("Android " + android.os.Build.VERSION.RELEASE);
        }
        if (call.method.equals("getNative")) {
            result.success("Get Android Success");
        }
        if (call.method.equals("sayHello")) {
            String message = call.argument("message");
            System.out.println("message: " + message);
            Map<String,Object> map = new HashMap<>();
            map.put("message",message);
            map.put("info","info-value");
            if(eventSink!=null){
                map.put("event","demoEvent");
                map.put("value","value is 10");
                eventSink.success(map);
            }
            result.success(map);
        } else {
            result.notImplemented();
        }
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        channel.setMethodCallHandler(null);
    }
}
